src/lib.rs

```diff
@@ -188,61 +203,106 @@ pub fn generate(mut args: Args) -> Result<()> {
     }
     resolve_favorite(&mut args)?;
 
-    let name = match args.name {
+    let project_name = match args.name {
         Some(ref n) => ProjectName::new(n),
         None if !args.silent => ProjectName::new(interactive::name()?),
         None => anyhow::bail!(
             "{} {} {}",
             emoji::ERROR,
             style("Project Name Error:").bold().red(),
-            style("Option `--silent` provided, but project name was not set. Please use `--project-name`.")
+            style("Option `--silent` provided, but project name was not set. Please use `--name`.")
                 .bold()
                 .red(),
         ),
     };
 
-    create_git(args, &name)?;
-    Ok(())
-}
+    let project_dir = create_project_dir(&project_name, args.force).ok_or_else(|| {
+        anyhow::anyhow!(
```

Comment:
same as above here, it's much more idiomatic to use `.context()?` instead of nesting like this